test/join.test.ts

```diff
@@ -10,6 +10,13 @@ describe("joinURL", () => {
     { input: ["/a"], out: "/a" },
     { input: ["a", "b"], out: "a/b" },
     { input: ["/", "/b"], out: "/b" },
+    { input: ["/a", "../b"], out: "/b" },
+    { input: ["../a", "../b"], out: "../b" },
```

Comment:
With URL behavior, base is always `/` (new URL('http://google.com/..').toString() ~> `'http://google.com/'`)

Result of similar join with URL:

```js
new URL('../b', 'http://test.com/../a').pathname 
// ~> /b
```

I expect to be consistent with URL behavior in ufo (instead of node path behavior)

Reply:
Of course. I was aiming to follow the behaviour above on L11 though. Should we update that to have a base too?

Reply:
In next major version we should always have leading slash as result of utils but it might also cause in usage regressions when they where depending on this behavior we might keep the behavior changes minimum to be on the safe side.

Reply:
I guess the benefit could be if people want to join URL segments which they then later join to something else. Preserving the remaining relativity could be beneficial in that case.

But I'm very happy to update, either way. Let me know which we should go for in this PR 🙏

Reply:
I think we should either not handle the resolution/normaliation at all (like current released version) or if we do, we make it URL compatible. If in some input cases only joinURL might contain .. it might be less predictable and even might lead to security issues when someone trusts join will resolve ..

Wdyt?

Reply:
(update from discussion): We preserve original `../` same as `./` for `joinURL` but use URL standard for `resolve`/`normalize`